src/flags/blocks.rs

```diff
@@ -144,6 +150,23 @@ impl Blocks {
             self.prepend_inode()
         }
     }
+
+    /// Tnserts a [Block] of variant [INode](Block::Label), if `self` does not already contain a
+    /// [Block] of that variant. The positioning will be best-effort approximation of coreutils
+    /// ls position for a security label
+    fn optional_insert_label(&mut self) {
```

Comment:
![2022-03-16-09-49-24](https://user-images.githubusercontent.com/14259816/158516423-9fbe3ca8-5800-491a-b994-9a4825808c6b.png)

We might have to join them together.

Reply:
This happens if dir content doesn't fit on one line and `--long` isn't used. As far as I'm concerned this is expected behavior with grid layout having `TopToBottom` direction. It has same output as coreutils if changed to `LeftToRight`. Is there a reason for that? I tried changing it and everything worked as expected but maybe I'm not creative enough when listing directories.

What do you mean by joining?  Putting context in `block_vec` along with name when rendering blocks? That wouldn't work e.g. for `lsd -Z --blocks size` or any other context+field combination.